src/daemon/daemon.cpp

```diff
@@ -3173,6 +3206,46 @@ mp::MountHandler::UPtr mp::Daemon::make_mount(VirtualMachine* vm, const std::str
                : vm->make_native_mount_handler(config->ssh_key_provider.get(), target, mount);
 }
 
+void mp::Daemon::configure_new_interfaces(const std::string& name, mp::VirtualMachine& vm, mp::VMSpecs& specs)
+{
+    std::vector<size_t> new_interfaces;
+
+    for (auto i = 0u; i < specs.extra_interfaces.size(); ++i)
+    {
+        auto& interface = specs.extra_interfaces[i];
+
+        if (interface.mac_address.empty()) // An empty MAC address means the interface needs to be configured.
+        {
+            new_interfaces.push_back(i);
+
+            interface.mac_address = generate_unused_mac_address(allocated_mac_addrs);
+        }
+    }
+
+    if (!new_interfaces.empty())
+    {
+        config->factory->prepare_networking(specs.extra_interfaces);
+
+        auto& commands = run_at_boot[name];
+
+        try
+        {
+            for (const auto i : new_interfaces)
+            {
+                vm.add_network_interface(i, specs.extra_interfaces[i]);
+
+                commands.push_back(generate_netplan_script(i, specs.extra_interfaces[i].mac_address));
+            }
+
+            commands.push_back("sudo netplan apply");
+        }
+        catch (const mp::NotImplementedOnThisBackendException&)
+        {
+            run_at_boot.erase(name);
```

Comment:
Shouldn't this rethrow? And actually, couldn't this be structured such that we only add `run_at_booy[name]` after (if) everything else succeeds?

Reply:
Rethrowing would mean to abort the execution of the `start` command. The problem with adding commands to the vector if everything succeeds would be possible, but the daemon would have to carry a lot of information about interfaces. Rolling back this way is much simpler.

Reply:
Oh I didn't realize this was called at start. We should instead detect that bridging isn't implemented when the user tries to set bridged=true and refuse it at that point. What happens now in that case?

For adding only after success, I was probably not very clear. What I mean is that this should compose the `commands` var but only add them to `run_at_boot` at the end, when we know everything succeeded. That would avoid having to catch and erase, or leaving lingering commands behind if other exceptions are thrown.